### PR TITLE
fix: allow for execution of proposals that have closed overrule proposal #ntrn-225

### PR DESCRIPTION
**Pull request by NeverHappened**
Rejected overrule proposals can be moved to closed state. That means we need to check for Closed state as well as rejected.

Task: https://hadronlabs.atlassian.net/browse/NTRN-225
Related PR's:
- https://github.com/neutron-org/neutron-dao/pull/100
- https://github.com/neutron-org/dao-ops/pull/27
- https://github.com/neutron-org/neutron-integration-tests/pull/268

### CODE REVIEW

**Comment by NeverHappened** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/8235688238

**Review by quasisamurai — APPROVED** [review]

File: contracts/subdaos/cwd-subdao-timelock-single/src/contract.rs [hunk 1]
```diff
@@ -333,13 +333,50 @@ pub fn query_proposal_execution_error(deps: Deps, proposal_id: u64) -> StdResult
...
+            migrated_ids.push(item.id.to_string());
+            PROPOSALS.save(deps.storage, item.id, &item)?;
+        }
+    }
+
+    Ok(Response::default().add_attribute("migrated_proposal_ids", migrated_ids.join(",")))
 }
 
-fn is_overrule_proposal_rejected(
+fn is_overrule_proposal_denied(
```
[the remark this answers is not in the document]
Comment (pr0n00gler) [reply contracts/subdaos/cwd-subdao-timelock-single/src/contract.rs:379]: Also let's add a comment what exactly this method checks, cause it's not quite clear naming actually

File: contracts/subdaos/cwd-subdao-timelock-single/src/contract.rs [hunk 2]
```diff
@@ -333,13 +333,50 @@ pub fn query_proposal_execution_error(deps: Deps, proposal_id: u64) -> StdResult
...
+            &OverruleQueryMsg::ProposalModule {},
+        )?;
+        let overrule_proposal: MainDaoProposalResponse = deps.querier.query_wasm_smart(
+            propose,
+            &MainDaoProposalModuleQueryMsg::Proposal {
+                proposal_id: overrule_proposal_id,
+            },
+        )?;
+
+        if overrule_proposal.proposal.status == Status::Closed {
```
Comment (pr0n00gler) [inline contracts/subdaos/cwd-subdao-timelock-single/src/contract.rs:369]: let's add a small comment what is exactly happening here. Like, we choose exactly `Closed` proposals and why set `Overruled` status for corresponding proposals in the timelock contract

**Comment by quasisamurai** [comment]
```
Caused by:
  package `wasmer-compiler-cranelift v4.2.6` cannot be built because it requires rustc 1.73 or newer, while the currently active rustc version is 1.71.0
```
please fix the CI issue

**Comment by NeverHappened** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/8451208399

**Pull request merged**